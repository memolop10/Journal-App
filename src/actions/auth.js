import { firebase,googleAuthProvider } from "../firebase/firebaseConfig";
import { types } from "../types/types"

export const startLoginEmailPassword = ( email, password ) => {
     return (dispatch) => {
         
         setTimeout(() => {
            dispatch(
              login( 123,'Pepe' )
            )
        }, 3500);
     }
}

export const startGoogleLogin = () => {
    return ( dispatch ) => {
          
               firebase.auth().signInWithPopup( googleAuthProvider )
                .then(({user}) => {
                    dispatch(
                        login(user.uid, user.displayName)
                    )
                    
                });
    }  
}


export const login = ( uid, displayName ) => {
    return {
        type: types.login,
        payload:{
            uid,
            displayName
        }
    }
}