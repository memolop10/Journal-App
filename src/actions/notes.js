import Swal from "sweetalert2";
import { db } from "../firebase/firebaseConfig";
import { loadNotes } from "../helpers/loadNotes";
import { types } from "../types/types";

export const startNewNote = () => {
    return async( dispatch, getState ) => {

        const uid = getState().auth.uid;
        console.log( uid )

        const newNote = {
            title:'',
            body:'',
            date: new Date().getTime()
        }

        const doc = await db.collection(`${ uid }/journal/notes`).add( newNote );
        console.log(doc)

        dispatch( activeMode( doc.id , newNote ) );
    }
}

export const activeMode = ( id, note ) => ({
    type: types.notesActive,
    payload:{
        id,
        ...note
    }
})

export const startLoadingNotes = ( uid ) => {
    return async( dispatch ) => {
        const notes = await loadNotes( uid );
        dispatch( setNotes( notes ) )
    }
}

export const setNotes = ( notes ) => ({
    type: types.notesLoad,
    payload: notes
})

export const startSaveNote = ( note ) => {
    return async( dispatch, getState ) => {

        const { uid } = getState().auth;

        if (!note.url) {
            delete note.url;
        }

        const noteToFirestore = { ...note };
        delete noteToFirestore.id;

        await db.doc(`${ uid }/journal/notes/${ note.id }`).update( noteToFirestore );

        dispatch( refreshNote( note.id, note ) )
        Swal.fire( 'Saved',note.title, 'success')
    }
}

export const refreshNote = ( id, note ) => ({
    type: types.notesUpdate,
    payload:{
        id, 
        note: {
            id,
            ...note
        }
    }
})