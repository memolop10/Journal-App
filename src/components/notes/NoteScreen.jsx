import React from 'react'
import NotesAppBar from './NotesAppBar'

const NoteScreen = () => {
    return (
        <div className="note__main-content">
            
            <NotesAppBar />

            <div className="notes__content">
                <input 
                    type="text" 
                    placeholder="Some awesome title"
                    className="notes__title-input"
                    autoComplete="off"
                />

                <textarea 
                    className="notes__textarea"
                    placeholder="What happend today" 
                    cols="30" 
                    rows="10"
                >

                </textarea>
            </div>

            <div className="notes__image">
                <img 
                    src="https://thumbs.dreamstime.com/z/estaci%C3%B3n-de-recreo-resplandor-azul-en-fondo-oscuro-182098130.jpg" 
                    alt="ps5"
                />
            </div>

        </div>
    )
}

export default NoteScreen
