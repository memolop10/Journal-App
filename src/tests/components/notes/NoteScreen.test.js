import React from "react";

import { mount } from "enzyme";
import { Provider } from "react-redux";
import '@testing-library/jest-dom'

import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk'
//para fingir rutas
import { MemoryRouter } from "react-router-dom/cjs/react-router-dom.min";
import { activeNote } from "../../../actions/notes";
import NoteScreen from "../../../components/notes/NoteScreen";


jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn(),
}))

const middlewares = [thunk];
const mockStore = configureStore( middlewares )

const initState = {
    auth:{
        uid:'1',
        name:'MEMO'
    },
    ui:{
        loading: false,
        msgError: null
    },
    notes:{
        active:{
            id:'ABC',
            title:'Hola',
            body:'Mundo',
            date: 0
        },
        notes:[]
    }
};

let store = mockStore( initState );
store.dispatch = jest.fn();

const wrapper = mount( 
    <Provider store = {store}>
        <MemoryRouter>
            <NoteScreen /> 
        </MemoryRouter>
    </Provider>
)



describe('Test on <NoteScreen/>', () => {
    test('Should display correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should dispatch the activeNote action', () => {
        wrapper.find('input[name="title"]').simulate('change',{
            target:{
                name:'title',
                value:'Hola de Nuevo'
            }
        })

        expect( activeNote ).toHaveBeenLastCalledWith(
            'ABC',
            {
                body:'Mundo',
                title: 'Hola de Nuevo',
                id: 'ABC',
                date: 0
            }
        )
    });
    
    
});
