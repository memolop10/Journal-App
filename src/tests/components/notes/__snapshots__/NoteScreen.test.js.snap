// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <NoteScreen/> Should display correctly 1`] = `
<div
  className="note__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 Agosto 2020
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hola"
    />
    <textarea
      className="notes__textarea"
      cols="30"
      name="body"
      onChange={[Function]}
      placeholder="What happend today"
      rows="10"
      value="Mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
